API/Data/Repositories/ChapterRepository.cs

```diff
@@ -247,4 +248,13 @@ public async Task<IList<MangaFile>> GetFilesForChaptersAsync(IReadOnlyList<int>
             .AsNoTracking()
             .ToListAsync();
     }
+
+    public async Task AddChapterProgress(int userId, ChapterDto chapter)
+    {
+        var progress = await _context.AppUserProgresses.FirstOrDefaultAsync(x =>
+            x.AppUserId == userId && x.ChapterId == chapter.Id);
+        if (progress == null) return;
+        chapter.PagesRead = progress.PagesRead;
+        chapter.ProgressLastModifiedUtc = progress.LastModifiedUtc;
```

Comment:
If you are adding progress, add the non Utc field as well to keep consistent.

Reply:
ok